ui/app/lib/satellizer.js

```diff
@@ -0,0 +1,856 @@
+        angular.forEach((keyValue || '').split('&'), function(keyValue) {
+          if (keyValue) {
+            value = keyValue.split('=');
+            key = decodeURIComponent(value[0]);
+            obj[key] = angular.isDefined(value[1]) ? decodeURIComponent(value[1]) : true;
+          }
+        });
+        return obj;
+      };
+
+      this.joinUrl = function() {
+        var joined = Array.prototype.slice.call(arguments, 0).join('/');
+
+        var normalize = function(str) {
+          return str
+            .replace(/[\/]+/g, '/')
+            .replace(/\/\?/g, '?')
+            .replace(/\/\#/g, '#')
+            .replace(/\:\//g, '://');
+        };
+
+        return normalize(joined);
+      };
+    })
+    .factory('satellizer.storage', ['satellizer.config', function(config) {
+      var storage = function(type) {
+        var stub = {
+          getItem: function(key) { return undefined; },
+          setItem: function(key, value) { return undefined; },
+          removeItem: function(key) { return undefined; }
+        };
+
+        switch (type) {
+          case 'localStorage':
+          case 'sessionStorage':
+            if (type in window && window[type] !== null) {
+              return window[type];
+            } else {
+              console.warn('Warning: Storage type `' + type + '` is disabled or unavailable. Satellizer will not work correctly.');
+              return stub;
+            }
+
+          default:
+            console.error('Error: Unexpected storage type `' + type + '`');
+            return stub;
+        }
+      };
+
+      return {
+        get: function(key) { return storage(config.storage).getItem(key); },
+        set: function(key, value) { return storage(config.storage).setItem(key, value); },
+        remove: function(key) { return storage(config.storage).removeItem(key); }
+      };
+    }])
+    .factory('satellizer.interceptor', [
+      '$q',
+      'satellizer.config',
+      'satellizer.storage',
+      function($q, config, storage) {
+        var tokenName = config.tokenPrefix ? config.tokenPrefix + '_' + config.tokenName : config.tokenName;
+        return {
+          request: function(httpConfig) {
+            var token = storage.get(tokenName);
+            if (token && config.httpInterceptor) {
+              if (config.authHeader && config.authToken) {
+                token = config.authToken + ' ' + token;
+              }
+              httpConfig.headers[config.authHeader] = token;
+            }
+            return httpConfig;
+          },
+          responseError: function(response) {
+            return $q.reject(response);
+          }
+        };
+      }])
+    .config(['$httpProvider', function($httpProvider) {
+      $httpProvider.interceptors.push('satellizer.interceptor');
+    }]);
+
+})(window, window.angular);
+
+// Base64.js Polyfill (@davidchambers)
+(function() {
+  var object = typeof exports != 'undefined' ? exports : this;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2260735530/issues/source?bid=3144114&fileBranchId=3144125#l822)